test/e2e/persistent_volumes.go

```diff
@@ -208,50 +318,62 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 		testPodSuccessOrFail(f, c, ns, checkPod)
 		checkPod = nil
 
-		// Delete the PersistentVolumeClaim
-		By("Deleting PersistentVolumeClaim to trigger PV Recycling")
-		framework.Logf("Deleting PersistentVolumeClaim to trigger PV Recycling")
-		err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
-		if err != nil {
-			framework.Failf("Delete PersistentVolumeClaim failed: %v", err)
-		}
-		// Wait for the PersistentVolume phase to return to Available
-		framework.Logf("Waiting for recycling process to complete.")
-		err = framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 3*time.Second, 300*time.Second)
-		if err != nil {
-			framework.Failf("Recycling failed: %v", err)
-		}
+		// Delete the PVC before deleting PV, wait for PV to be Available
+		deletePVCandValidatePV(c, ns, pvc, pv)
 
-		// Examine the PersistentVolume.ClaimRef and UID. Expect nil values.
-		pv, err = c.PersistentVolumes().Get(pv.Name)
-		if pv.Spec.ClaimRef != nil && len(pv.Spec.ClaimRef.UID) > 0 {
-			crjson, _ := json.MarshalIndent(pv.Spec.ClaimRef, "", "  ")
-			framework.Failf("Expected a nil pv.ClaimRef or empty UID. Found: ", string(crjson))
-		}
-
-		// Delete the PersistentVolume
-		By("Deleting PersistentVolume")
+		// Last cleanup step is to delete the pv
 		deletePersistentVolume(c, pv)
```

Comment:
As the tests use `Failf` a lot, any created PV will be leaked after failed test. You should call `defer deletePersistentVolume(c, pv)` just after `createPVandPVC` (+ the same in the test below).